launcher/LoggedProcess.cpp

```diff
@@ -59,14 +59,18 @@ LoggedProcess::~LoggedProcess()
     }
 }
 
-QStringList reprocess(const QByteArray & data, QString & leftover)
+QStringList reprocess(const QByteArray& data, QByteArray& leftover)
 {
-    QString str = leftover + QString::fromLocal8Bit(data);
```

Comment:
I want to keep the data as a byte array until we definitely can parse it as UTF-8. I was concerned `leftover` might contain invalid fragments of UTF-8 if the process's std out was flushed mid-line, so I split on the `\n` byte and any bytes after that are saved for the next batch of data.

Reply:
From the docs of `QString::fromUtf8`:
> UTF-8 is a Unicode codec and can represent all characters in a Unicode string like [QString](https://doc.qt.io/qt-6/qstring.html). However, invalid sequences are possible with UTF-8 and, if any such are found, they will be replaced with one or more "replacement characters", or suppressed. These include non-Unicode sequences, non-characters, overlong sequences or surrogate codepoints encoded into UTF-8.

Reply:
I see... But then, to avoid having to deal with that weird `QByteArray` interface, which clogs up the code a bit, couldn't we perhaps create a `QString` based on the `QByteArray` and use its interface as we were doing before? Or maybe use something like [`QTextDecoder`](https://doc.qt.io/qt-5/qtextdecoder.html) to avoid having to deal with that leftover at all.

I just would like to get rid of all that `truncate` `remove_if` `transform` stuff if possible, it looks kinda convoluted and i feel like it will come bite us later xD

Reply:
`QTextDecoder` looks like it does exactly what I'm doing here manually, so I'll migrate to just using that!

> If not all characters are used (e.g. if only part of a multi-byte encoding is at the end of the characters), the decoder remembers enough state to continue with the next call to this function.